File: backend/src/main/java/com/logistics/enums/FeeConfiguration.java
package com.logistics.enums;

public class FeeConfiguration {
    public enum FeeType {
        COD, // Phí thu hộ
        PACKAGING, // Phí đóng gói khi tạo đơn hàng tại quầy
        INSURANCE, // Phí bảo hiểm cho orderValue
        VAT, // Thuế VAT
    }

    public enum CodFeeType {
        FIXED, // Phí cố định
        PERCENTAGE // Phí theo %
    }
}

File: frontend/src/types/feeConfiguration.ts
export interface FeeConfiguration {
  id: number;
  serviceTypeId?: number;
  serviceTypeName?: string;
  feeType: string; // COD | PACKAGING | INSURANCE | VAT
  calculationType: string; // FIXED | PERCENTAGE
  feeValue: number;
  minOrderFee?: number;
  maxOrderFee?: number;
  active: boolean;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateFeeConfigurationPayload {
  serviceTypeId?: number;
  feeType: string; // COD | PACKAGING | INSURANCE | VAT
  calculationType: string; // FIXED | PERCENTAGE
  feeValue: number;
  minOrderFee?: number;
  maxOrderFee?: number;
  active?: boolean;
  notes?: string;
}

export interface QueryState {
  page: number;
  limit: number;
  search: string;
  feeType?: string;
  serviceTypeId?: number;
  active?: boolean;
}

export interface Option {
  label: string;
  value: string;
}

export interface FeeConfigurationsTableProps {
  loading: boolean;
  rows: FeeConfiguration[];
  onView: (record: FeeConfiguration) => void;
  onEdit: (record: FeeConfiguration) => void;
  onDelete: (id: number) => void;
  feeTypeLabel: (value: string) => string;
}

export type UpdateFeeConfigurationPayload = Partial<CreateFeeConfigurationPayload>;
